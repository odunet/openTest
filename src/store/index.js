import React, { useReducer, createContext } from 'react';
import reducer from '../reducers';

const initialState = {
  title: 'Title of Widget',
  weather: {
    image: '',
    celsiusTemp: '',
    fahrenheitTemp: '',
    city: '',
    wind: '',
  },
  isCelsius: true,
  showWind: true,
};

const store = createContext(initialState);

const { Provider } = store;

const StateProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);
  return <Provider value={{ state, dispatch }}>{children}</Provider>;
};

export { store, StateProvider };
