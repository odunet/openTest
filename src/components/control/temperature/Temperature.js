import React, { useState, useContext } from 'react';
import { TextLabel, Radio, FlexColDiv } from '../../../shared/styledComponents';
import { changeUnits_Action } from '../../../actions';
import { store } from '../../../store';
import { handleCheck } from '../../../helper/handleCheck';

export default function Temperature() {
  const [radio, setRadio] = useState('C');

  // Bring in dispatch
  const { dispatch } = useContext(store);

  return (
    <>
      <TextLabel htmlFor='controlTitle'>Temperature</TextLabel>
      <FlexColDiv>
        <FlexColDiv>
          <Radio
            type='radio'
            data-testid='celsuis'
            id='celcuis'
            name='temperature'
            value='C'
            onChange={(e) =>
              handleCheck(dispatch, setRadio, changeUnits_Action, e.target)
            }
            checked={radio === 'C'}
          />
          <p>
            <sup>o</sup>C
          </p>
        </FlexColDiv>
        <FlexColDiv>
          <Radio
            type='radio'
            data-testid='fahrenhiet'
            id='fahrenhiet'
            name='temperature'
            value='F'
            onChange={(e) =>
              handleCheck(dispatch, setRadio, changeUnits_Action, e.target)
            }
            checked={radio === 'F'}
          />
          <p>F</p>
        </FlexColDiv>
      </FlexColDiv>
    </>
  );
}
