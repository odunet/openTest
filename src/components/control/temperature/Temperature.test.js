import { render, screen } from '@testing-library/react';
import Temperature from './Temperature';
import toJson from 'enzyme-to-json';
import Enzyme, { shallow, mount } from 'enzyme';
import { useContext } from 'react';

//React testing library
import { fireEvent, cleanup } from '@testing-library/react';
import { store } from '../../../store';

describe('Render Temperature UI', () => {
  test('renders Temperature', () => {
    render(<Temperature />);
    const temperatureElement = screen.getByText(/Temperature/i);
    expect(temperatureElement).toBeInTheDocument();
  });
});

describe('Snapshot', () => {
  test('Should take a snapshot of Temperature component', () => {
    const container = shallow(<Temperature />);
    expect(toJson(container)).toMatchSnapshot();
  });
});

jest.mock('../../../helper/handleCheck.js', () => {
  return {
    handleCheck: (...args) => {
      return;
    },
  };
});

describe('Check checkbox change', () => {
  test('Checkbox is working', () => {
    render(<Temperature />);

    // Click button
    fireEvent.click(screen.getByTestId('fahrenhiet'));
    expect(screen.getByTestId('fahrenhiet').value).toEqual('F');
  });
});
