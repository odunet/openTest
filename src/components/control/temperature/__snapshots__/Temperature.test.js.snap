// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Should take a snapshot of Temperature component 1`] = `
<Fragment>
  <styled.label
    htmlFor="controlTitle"
  >
    Temperature
  </styled.label>
  <styled.div>
    <styled.div>
      <styled.input
        checked={true}
        data-testid="celsuis"
        id="celcuis"
        name="temperature"
        onChange={[Function]}
        type="radio"
        value="C"
      />
      <p>
        <sup>
          o
        </sup>
        C
      </p>
    </styled.div>
    <styled.div>
      <styled.input
        checked={false}
        data-testid="fahrenhiet"
        id="fahrenhiet"
        name="temperature"
        onChange={[Function]}
        type="radio"
        value="F"
      />
      <p>
        F
      </p>
    </styled.div>
  </styled.div>
</Fragment>
`;
